src/cargo/core/compiler/build_config.rs

```diff
@@ -68,7 +68,12 @@ impl BuildConfig {
                  its environment, ignoring the `-j` parameter",
             )?;
         }
-        let jobs = jobs.or(cfg.jobs).unwrap_or(::num_cpus::get() as u32);
+        let jobs = match jobs.or(cfg.jobs) {
+            None => ::num_cpus::get() as u32,
+            Some(j) if j > 0 => j,
+            Some(j) if j == 0 => anyhow::bail!("jobs may not be 0"),
+            Some(_) => anyhow::bail!("jobs may not have a negative value"),
```

Comment:
This case isn't possible, since it is an unsigned integer, is it?  Perhaps this could be restructured to be a simple `let` statement as before, and then a conditional `if jobs == 0 {bail!("...")}`?

Reply:
Sure, that makes sense. I will review and edit.